# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLShader.cpp
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLShader.cpp
--- a/dom/canvas/WebGLShader.cpp	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLShader.cpp	Wed Jan 08 22:19:23 2020 +0000
@@ -80,10 +80,12 @@
   mContext->gl->fDeleteShader(mGLName);
 }
 
-void WebGLShader::ShaderSource(const std::string& source) {
-  if (!ValidateGLSLPreprocString(mContext, source)) return;
-
-  mSource = source;
+void WebGLShader::ShaderSource(const std::string& cleanSource) {
+  const auto badChar =
+      CheckGLSLPreprocString(mContext->IsWebGL2(), cleanSource);
+  MOZ_ASSERT(!badChar);
+  if (badChar) return;
+  mSource = cleanSource;
 }
 
 void WebGLShader::CompileShader() {